### DISCUSSION TITLE
[Art competition] Abstract Interpretation

### DISCUSSION THREAD

**Discussion by oojahooo** [prosyslab-classroom/cs524-program-analysis#271] [Art competition]
![DALL·E 2022-11-25 14 40 11](https://user-images.githubusercontent.com/20193799/203908895-a24a52ec-84f0-4b32-85fa-bb5507cc1b3f.png)

## Short Description
For me, [Abstract Interpretation] means [Buddha's palm that Wukong cannot escape]".

## Long Description (up to 3 sentences)
Abstract interpretation is a powerful framework that guarantees soundness. The "guarantees soundness" means that abstract semantics can check all of possible program executions within a finite computation even if a program has infinite concrete states with high variance. This is very powerful and significant concept in program analysis.

**Comment by KihongHeo** [comment]
Good. Interesting idea.

Can we have another analogy that is more positive and proactive? Abstract semantics does not limit the capability of concrete semantics.

**Comment by oojahooo** [comment]
That seems to be an important point. Do you know zorbing? I think it is another good analogy of abstract interpretation. A person in a zorb ball can go anywhere and the zorb ball always covers the person. But the zorb ball should be larger than the person so it can hit an obstacle even though the person is in a position not to actually touch the obstacle.

![image](https://user-images.githubusercontent.com/20193799/204678845-b9a8585a-05bd-484c-8d28-14293e4bf670.png)
